src/main/java/homework_3/ImmutableWorker.java

```diff
@@ -0,0 +1,114 @@
+package homework_3;
+
+import jdk.nashorn.internal.ir.annotations.Immutable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * In general immutable class means that once an object is created, we cannot change its content, so:
+ * Immutable class should be final.
+ * Its fields should be private final.
+ * There should be getters but no setters.
+ * For link objects there should be a deep copy in the constructor.
+ */
+
+public final class ImmutableWorker {
+    private final String name;
+    private final String department;
+    private final int id;
+    private final List<String> tasks;
+    private final Age age;
+
+    public ImmutableWorker(String name, String department, int id, List<String> tasks, Age age) {
+        this.name = name;
+        this.department = department;
+        this.id = id;
+        this.tasks = new ArrayList<>(tasks);
+        this.age = new Age(age.getDay(), age.getMonth(), age.getYear());
+    }
+
+    public ImmutableWorker(String name, String department, int id, Age age) {
+        this.name = name;
+        this.department = department;
+        this.id = id;
+        this.tasks = new ArrayList<>();
+        this.age = new Age(age.getDay(), age.getMonth(), age.getYear());
+    }
+
+    public ImmutableWorker(String name, int id, Age age) {
+        this.name = name;
+        this.department = "No department";
+        this.id = id;
+        this.tasks = new ArrayList<>();
+        this.age = new Age(age.getDay(), age.getMonth(), age.getYear());
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDepartment() {
+        return department;
+    }
+
+    public int getId() {
+        return id;
+    }
+
+    public List<String> getTasks() {
+        return tasks;
+    }
+
+    public Age getAge() {
```

Comment:
ImmutableWorker is not immutable. I can getAge.setDay(5);

Reply:
I removed setters from the Age Class as well